File: apps/android/app/src/main/java/com/velix/app/core/AppSession.kt
package com.velix.app.core

import androidx.compose.runtime.compositionLocalOf

class AppSession(val instanceStore: InstanceStore) {
    fun apiClient(instance: Instance): ApiClient = ApiClient(instance.baseUrl, instance.accessToken)

    val activeApiClient: ApiClient?
        get() = instanceStore.activeInstance.value?.let { apiClient(it) }
}

val LocalAppSession = compositionLocalOf<AppSession> { error("AppSession não fornecida") }

File: apps/ios/Velix/Core/AppSession.swift
import Foundation
import Observation

@Observable
final class AppSession {
    let instanceStore: InstanceStore

    init(instanceStore: InstanceStore = InstanceStore(keychain: KeychainStore(service: "com.velix.app"))) {
        self.instanceStore = instanceStore
    }

    var hasAnyInstance: Bool {
        !instanceStore.instances.isEmpty
    }

    func apiClient(for instance: Instance) -> APIClient {
        APIClient(baseURL: instance.baseURL, token: instance.accessToken)
    }

    var activeAPIClient: APIClient? {
        guard let active = instanceStore.activeInstance else { return nil }
        return apiClient(for: active)
    }
}
